lib/associations/belongs-to-many.js

```diff
@@ -464,6 +465,10 @@ BelongsToMany.prototype.injectGetter = function(obj) {
       }
     }
 
+    if (options.hasOwnProperty('schema')) {
+      model = model.schema(options.schema, options.schemaDelimiter);
+    }
```

Comment:
They could but then that would require the user to setup a separate association for every schema that they need to access this association in with different names. In an ideal world I can setup one set of models and associations and use them across all schemas without having to define the models and associations again with different names for each schema. That being said it would work as you described as well if that use case make more sense for the user than you would just not pass the schema option and it would use the schema associated with the model when the association was setup.